logstash-core/lib/logstash/plugins/registry.rb

```diff
@@ -151,13 +151,14 @@ def load_available_plugins
       require "logstash/plugins/builtin"
 
       GemRegistry.logstash_plugins.each do |plugin_context|
-        if plugin_context.spec.metadata.key?('java_plugin')
-          jar_files = plugin_context.spec.files.select {|f| f =~ /.*\.jar/}
+        if plugin_context.spec.metadata.key?('java_plugin') && LogStash::SETTINGS.get_value('pipeline.plugin_classloaders')
+          # Find *.jar file from the require path
+          jar_files = Dir.glob(plugin_context.spec.full_require_paths.map{|path| "#{path}/**/*.jar"})
```

Comment:
`Gem::Specification#matches_for_glob` is based on the require path and simplifies this a bit :)

```suggestion
          jar_files = plugin_context.spec.matches_for_glob("**/*.jar")
```

Reply:
Gah, completely missed that method. Nice catch